puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -0,0 +1,136 @@
+def ruby = '2.4'
+def katello_versions = [
+    'master': 'develop',
+    'KATELLO-3.6': '1.17-stable',
+    'KATELLO-3.5': '1.16-stable'
+]
+
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup Git Repos') {
+            steps {
+                checkout changelog: true, poll: false, scm: [
+                    $class: 'GitSCM',
+                    branches: [[name: '${ghprbActualCommit}']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                    userRemoteConfigs: [
+                        [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/katello/katello']
+                    ]
+                ]
+
+                dir('foreman') {
+                    git url: "https://github.com/theforeman/foreman", branch: katello_versions[ghprbTargetBranch]
+                }
+            }
+        }
+        stage("Setup RVM") {
+            steps {
+
+                configureRVM(ruby)
+
+            }
+        }
+        stage('Configure Environment') {
+            steps {
+
+                dir('foreman') {
+                    addSettings([
+                        organizations: true,
+                        locations: true
+                    ])
+                    addGem()
+                    databaseFile(gemset())
+                }
+
+            }
+        }
+        stage('Configure Database') {
+            steps {
+
+                dir('foreman') {
+
+                    configureDatabase(ruby)
+
+                }
+
+            }
+        }
+        stage('Run Tests') {
+            steps {
+
+                parallel (
+                    'tests': {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake jenkins:katello TESTOPTS="-v"'], ruby)
+                        }
+                    },
+                    'rubocop': {
+                        dir('foreman') {
+                            withRVM(['bundle exec rake katello:rubocop TESTOPTS="-v" --trace'], ruby)
+                        }
+                    },
+                    'react-ui': {
```

Comment:
can we mark this step as skipped when there is no `package.json` available? (if that's not too complicated, as it's just a visual thing I'd love to have)